scipy/stats/_stats.pyx

```diff
@@ -0,0 +1,31 @@
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+#cython: cdivision=True
+from libc cimport math
+
+
+ctypedef fused real:
+    short
+    int
+    long
+    float
+    double
+
+
+cpdef double ks_2samp(real[:] data1, real[:] data2):
```

Comment:
I mostly made it a cpdef so that I can annotate the return type as double (mostly for documentation purposes -- "def" functions can't have a return type annotated, I think).